### PR TITLE
chore: remove remaining Objective C code from WireSystem - WPB-10368

### PR DESCRIPTION
**Pull request by caldrian**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10368" title="WPB-10368" target="_blank">TaskWPB-10368</a>  Migrate parts of the code to SPM
  </td></table>
  <br />
 

### Issue

Swift packages cannot have mixed Swift and Objective C code.
In order to prepare a future switch to a WireSystem Swift package, this PR removes the remaining Objective C code:
- moves the code from ZMSLogging.m into macros in ZMSLogging.h
- converts ZMLogLevel to Swift
- rewrites `ZMAssertionDump` in Swift (ZMAssertionDumpFile.swift)
- converts `ZMSAssertionTests.m` to `ZMAssertionDumpFileTests.swift`

all tests pass: https://github.com/wireapp/wire-ios/actions/runs/10074940417/job/27852138218

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [x] Make sure you use the API for UI elements that support large fonts.
- [x] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [x] New UI elements have Accessibility strings for VoiceOver.

### CODE REVIEW

File: wire-ios-system/Source/ZMSLog.swift [hunk 1]
```diff
@@ -245,30 +258,28 @@ extension ZMSLog {
...
 // MARK: - Save on disk & file management
 
-extension ZMSLog {
+public extension ZMSLog {
     private enum Constant {
         static let maxNumberOfLogFiles = 5
     }
 
-    static var cachesDirectory: URL? {
+    internal static var cachesDirectory: URL? {
```
Comment (samwyndham) [inline wire-ios-system/Source/ZMSLog.swift:266]: How come the need for an _explicit_ `internal` here and in some other places?

File: wire-ios-system/Source/ZMAssertionDumpFile.swift [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+public final class AssertionDumpFile: NSObject {
+
+    @available(*, unavailable)
+    override public init() {
+        fatalError()
+    }
+
+    public static var url: URL {
+        get throws {
+            let applicationSupportDirectory = try FileManager.default.url(
```
Comment (samwyndham) [inline wire-ios-system/Source/ZMAssertionDumpFile.swift:31]: It seems like a big side effect to getting a URL but I guess in practice this directory exists right? And in iOS 16 we will be able to replace it with [URL.applicationSupportDirectory ](https://developer.apple.com/documentation/foundation/url/3988452-applicationsupportdirectory)... Just thinking out loud. No need to change anything.

**Review by samwyndham — APPROVED** [review]
Nice work!

**Review by agisilaos — APPROVED** [review]

File: wire-ios-system/Source/ZMSLog.swift (hunk 1 above)
Comment (caldrian) [reply wire-ios-system/Source/ZMSLog.swift:266]: There is this SwiftFormat rule [extensionAccessControl](https://github.com/nicklockwood/SwiftFormat/blob/main/Rules.md#extensionAccessControl) and we agreed on having an access modifier next to `extension`. 
In this case the extension is `public`, so any declaration that shouldn't be public needs `internal`.
The formatter did this automatically.

File: wire-ios-system/Source/ZMAssertionDumpFile.swift (hunk 2 above)
Comment (caldrian) [reply wire-ios-system/Source/ZMAssertionDumpFile.swift:31]: I agree. 
Feel free to change it in another iteration.

File: wire-ios-system/Source/ZMSLog.swift (hunk 1 above)
Comment (samwyndham) [reply wire-ios-system/Source/ZMSLog.swift:266]: Oh yeah, I see! Thank you!

**Pull request merged**